staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -430,10 +430,18 @@ func AddDryRunFlag(cmd *cobra.Command) {
 	cmd.Flags().Lookup("dry-run").NoOptDefVal = "unchanged"
 }
 
+func defaultFieldManager(cmd *cobra.Command) string {
+	return fmt.Sprintf("kubectl-%s", strings.ReplaceAll(cmd.CommandPath(), " ", "-"))
```

Comment:
Ah! That's actually a good idea!

Reply:
I had to abandon this idea because the `*cobra.Command` isn't a child in the full command path yet.

I found this when trying this for `kubectl set env`: the cobra command path resulted in the field manager `kubectl-env` instead of `kubectl-set-env`.

Instead, I updated the change to just set the default field manager explicitly when adding the flag e.g. `	cmdutil.AddFieldManagerFlagVar(cmd, &o.fieldManager, "kubectl-set-env")`